libs/antsi/src/lib.rs

```diff
@@ -1,14 +1,500 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+// Good reference to begin with: https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797
+#![no_std]
+#![cfg_attr(
+    nightly,
+    feature(provide_any, error_in_core, error_generic_member_access)
+)]
+#![cfg_attr(all(doc, nightly), feature(doc_auto_cfg))]
+#![warn(
+    missing_docs,
+    unreachable_pub,
+    clippy::pedantic,
+    clippy::nursery,
+    clippy::undocumented_unsafe_blocks,
+    clippy::dbg_macro,
+    clippy::print_stdout,
+    clippy::print_stderr,
+    clippy::alloc_instead_of_core,
+    clippy::std_instead_of_alloc,
+    clippy::std_instead_of_core,
+    clippy::if_then_some_else_none
+)]
+#![allow(clippy::redundant_pub_crate)] // This would otherwise clash with `unreachable_pub`
+#![allow(clippy::module_name_repetitions)]
+#![allow(missing_docs)]
```

Comment:
I'd say we remove this and put it in appropriate places (or just add the docs directly) instead.